### PR TITLE
use custom subjects as needed for one-liners

### PR DESCRIPTION
**Pull request by graysonwright**
See #250 

This PR encourages the use of custom subjects for one-line specs as opposed to breaking the specs into multiple lines.

### CODE REVIEW

**Comment by gylaz** [comment]
My concern here is that it loosens our recommendation about avoiding `subject` in general. We **only** want to use it in a very specific case, that is less that 10% (totally arbitrary number) of test that we write.

What if we leave `subject` in `best-practices`, and treat the line in `style` as an amendment?

File: best-practices/README.md [hunk 1]
```diff
@@ -107,7 +107,8 @@ Testing
 -------
 
 * Avoid `any_instance` in rspec-mocks and mocha. Prefer [dependency injection].
-* Avoid `its`, `let`, `let!`, `specify`, `before`, and `subject` in RSpec.
+* Avoid `its`, `let`, `let!`, `specify`, and `before` in RSpec.
+* Avoid using `subject` explicitly in RSpec `it` blocks.
```
Comment (graysonwright) [inline best-practices/README.md]: @gylaz does this extra line help to clarify at all?

Comment (gylaz) [inline best-practices/README.md]: I'm not sure that the problem is people using `subject` in `it` blocks -- is that even possible?

How about something like this?
```
Avoid using `subject` in RSpec, except for groups of one-line specs.
```

**Comment by gylaz** [comment]
With the example it makes sense! Does it make sense to add that example somewhere?

Or perhaps reward to use `inside`: "Avoid using `subject` explicitly __inside of an__ RSpec `it` block."

**Comment by graysonwright** [comment]
@gylaz @mcmire changes made, one last look?

**Comment by gylaz** [comment]
LGTM

**Comment by mcmire** [comment]
:star:

**Comment by graysonwright** [comment]
Dammit! I dissociated the code from this PR again!
:angry:

Figured out why, though. Creating aliases now so it doesn't happen again.
Merged into master in c694bd0

**Pull request closed without merging**